spec/errors.md

```diff
@@ -241,49 +241,71 @@ or for private implementation use that is not supported by the current implement
 > * {{The value is not one.}}
 > ```
 
-### Invalid Expression
+### Unsupported Statement
+
+An **_<dfn>Unsupported Statement</dfn>_** error occurs when a message includes a _reserved statement_.
+
+> For example, attempting to format this message
+> would always result in an _Unsupported Statement_ error:
+>
+> ```
+> .some {|horse|}
+> {{The message body}}
+> ```
 
-An **_<dfn>Invalid Expression</dfn>_** error occurs when a _message_ includes an _expression_
-whose implementation-defined internal requirements produce an error during _function resolution_
-or when a _function_ returns a value (such as `null`) that the implementation does not support.
+## Message Function Errors
 
-An **_<dfn>Operand Mismatch Error</dfn>_** is an _Invalid Expression_ error that occurs when
+A **_<dfn>Message Function Error</dfn>_** is any error that occurs
+when calling a message function implementation
+or which depend on validation associated with a specific function.
+
+Implementations SHOULD provide a way for custom functions to specify
+which of the following errors ought to be emitted,
+as well as allowing for other implementation-defined _Message Function Errors_.
```

Comment:
Nit-picking the language here a bit. I got rid of "custom" because it applies to non-custom functions also.

```suggestion
Implementations SHOULD provide a way for _functions_ to emit 
(or cause to be emitted) any of the types of error defined in this section.
Implementations MAY also provide implementation-defined _Message Function Error_ types.
```